gdal2xyz.py - improvements and added documentation. (#3416)
OSGeo/gdal: gdal/doc/source/programs/gdal2xyz.rst

.. _gdal2xyz:

================================================================================
gdal2xyz
================================================================================

.. only:: html

    Translates a raster file into `xyz` format.

.. Index:: gdal2xyz

Synopsis
--------

.. code-block::

    gdal2xyz [-help]
        [-skip factor]
        [-srcwin xoff yoff xsize ysize]
        [-b band]* [-allbands]

>         [-csv]

Reviewer: I believe `skipnodata` is missing

        [-srcnodata value] [-dstnodata value]
        src_dataset [dst_dataset]

Description
-----------

The :program:`gdal2xyz` utility can be used to translate a raster file into xyz format.
`gdal2xyz` can be used as an alternative to `gdal_translate of=xyz`, but supporting other options,
for example:

    * Select more then one band
    * Skip or replace nodata value
    * Return the output as numpy arrays.

.. program:: gdal2xyz

.. option:: -skip

    How many rows/cols to skip in each iteration.

.. option:: -srcwin <xoff> <yoff> <xsize> <ysize>

    Selects a subwindow from the source image for copying based on pixel/line location.

.. option:: -b, -band <band>

    Select band *band* from the input spectral bands for output.
    Bands are numbered from 1 in the order spectral bands are specified.
    Multiple **-b** switches may be used.
    When no -b switch is used, the first band will be used.
    In order to use all input bands set `-allbands` or `-b 0`.

.. option:: -allbands

    Select all input bands.

.. option:: -csv

    Use comma instead of space as a delimiter.

.. option:: -skipnodata

>     Exclude from the output lines with nodata value (as determinate by srcnodata)

Reviewer: ```suggestion
    Exclude the output lines with nodata value (as determined by srcnodata)
```

.. option:: -srcnodata

>     Determinants what is the nodata value of the dataset (for skipping or replacing)

Reviewer: ```suggestion
    The nodata value of the dataset (for skipping or replacing)
```

.. _gdal2xyz:

================================================================================
gdal2xyz
================================================================================

> 

Reviewer 2: ```suggestion

.. versionadded:: 3.3

```

.. only:: html

    Translates a raster file into `xyz` format.

.. Index:: gdal2xyz

Synopsis
--------

.. code-block::

    gdal2xyz [-help]
        [-skip factor]
        [-srcwin xoff yoff xsize ysize]
        [-b band]* [-allbands]
        [-skipnodata]
        [-csv]
        [-srcnodata value] [-dstnodata value]
        src_dataset [dst_dataset]

Description
-----------

The :program:`gdal2xyz` utility can be used to translate a raster file into xyz format.
`gdal2xyz` can be used as an alternative to `gdal_translate of=xyz`, but supporting other options,
for example:

    * Select more then one band
    * Skip or replace nodata value
    * Return the output as numpy arrays.

.. program:: gdal2xyz

> .. option:: -skip

Reviewer 2: ```suggestion
.. option:: -skip <factor>
```

.. _gdal2xyz:

================================================================================
gdal2xyz
================================================================================

.. only:: html

    Translates a raster file into xyz format.

.. Index:: gdal2xyz

Synopsis
--------

.. code-block::

    gdal2xyz [-help]
        [-skip factor]
        [-srcwin xoff yoff xsize ysize]
        [-b band]* [-allbands]
        [-csv]
        [-srcnodata value] [-dstnodata value]
        src_dataset [dst_dataset]

Description
-----------

> The :program:`gdal2xyz` utility can be used to translates a raster file into xyz format.

Reviewer 3: `s/translates/translates/`

Author: done

> gdal2xyz is an alternative to `gdal_translate` with `-of=xyz` with other options,

Reviewer 3: backticks around `gdal2xyz`

"with other" -> "with other options" ?

Author: I've rephrased that.

> like selecting more then one band, skip or replace nodata value, and return output as numpy arrays.

Reviewer 3: "like selecting" -> "for example, to select"

Author: done

    In order to use all input bands set `-allbands` or `-b 0`.

.. option:: -allbands

    Select all input bands.

.. option:: -csv

    use comma instead of space as a delimiter

.. option:: -skipnodata

    Exclude from the output lines with nodata value (as determinate by srcnodata)

.. option:: -srcnodata

    Determinants what is the nodata value of the dataset (for skipping or replacing)
    default (`None`) - Use the dataset NoDataValue;
    `Sequence`/`Number` - use the given nodata value (per band or per dataset).

.. option:: -dstnodata

    Replace source nodata with a given nodata. Has an effect only if not setting `-skipnodata`
    default(`None`) - use srcnodata, no replacement;
    `Sequence`/`Number` - replace the `srcnodata` with the given nodata value (per band or per dataset).

.. option:: -h, --help

    Show help message and exit.

.. option:: <src_dataset>

    The source dataset name. It can be either file name, URL of data source or
    subdataset name for multi-dataset files.

.. option:: <dst_dataset>

    The destination file name.

Examples
--------

::

    gdal2xyz -b 1 -b 2 -dstnodata 0 input.tif output.txt

> To create a text in xyz format of input.tif, including bands 1 and 2, while replacing the dataset nodata values with zeros.

Reviewer 3: backticks around `input.tif`

Author: done
